migration_helpers/resources.py

```diff
@@ -2,19 +2,30 @@
 from .cpu import transform_cpu, CPU_COUNT
 
 
-def migrate_resources(resources):
-    # Handle empty resources, with sane defaults
-    if not resources:
-        return {
-            "limits": {
-                "cpus": (CPU_COUNT or 2) / 2,
-                "memory": f"{TOTAL_MEM / 1024 / 1024}M",
-            }
-        }
+def migrate_resources(resources, gpus=None):
+    gpus = gpus or {}
 
-    return {
+    result = {
         "limits": {
-            "cpus": transform_cpu(resources.get("limits", {}).get("cpu", "")),
-            "memory": transform_memory(resources.get("limits", {}).get("memory", "")),
+            "cpus": (CPU_COUNT or 2) / 2,
+            "memory": {TOTAL_MEM / 1024 / 1024},
         }
     }
+
+    if resources.get("limits", {}).get("cpu", ""):
+        result["limits"].update(
+            {"cpus": transform_cpu(resources.get("limits", {}).get("cpu", ""))}
+        )
+    if resources.get("limits", {}).get("memory", ""):
+        result["limits"].update(
+            {"memory": transform_memory(resources.get("limits", {}).get("memory", ""))}
+        )
+
+    for gpu in gpus.items() if gpus else []:
```

Comment:
Curious about `gpus` format here - can you share it please?

Reply:
So now it won't migrate nvidia ones if i am not wrong?

Reply:
Correct, one way to solve this is to pass information about nvidia gpus in the migration script.
if there was previously `x` number of nvidia gpus, we can pick the first `x` nvidia gpus available at the time of migration. But in this case the UUIDs values etc  are needed in the migration script

Reply:
I think just passing through one nvidia gpu regardless of the count is fine if a nvidia gpu earlier was configured

Reply:
@stavros-k the migration can provide list of resources available on that end so the migration can go through seamlessly.
@sonicaj something like the following can work perhaps?
```
{'nvidia_gpus': ['uuid1', 'uuid2']}
```

Reply:
It can be passed through as additional context so the migration can take an informed decision - @stavros-k would there be other cases which the config migration might require?

Reply:
I can't think of something else currently, neither found something else missing!

Reply:
> ```
> 
> {'nvidia_gpus': ['uuid1', 'uuid2']}
> 
> ```

The pci_slot would be needed too. Otherwise the new schema won't validate